app/javascript/vue/repository_item_sidebar/repository_values/HighlightComponent.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+  <div class="flex gap-3">
+    <div id="navigation-text">
+      <div class="flex flex-col py-2 px-0 gap-3 self-stretch w-[130px] h-[130px]  justify-center items-center">
+        <div class="flex flex-col w-[130px] h-[130px] justify-between text-right">
+          <div @click="handleSideNavClick" id="text-item-1" class="hover:cursor-pointer text-sn-grey"
+            :class="{ 'text-sn-science-blue': selectedNavText === 'text-item-1' }">{{
+              i18n.t('repositories.highlight_component.information_label') }}
+          </div>
+          <div @click="handleSideNavClick" id="text-item-2" class="hover:cursor-pointer text-sn-grey"
+            :class="{ 'text-sn-science-blue': selectedNavText === 'text-item-2' }">{{
+              i18n.t('repositories.highlight_component.custom_columns_label') }}
+          </div>
+          <div @click="handleSideNavClick" id="text-item-3" class="hover:cursor-pointer text-sn-grey"
+            :class="{ 'text-sn-science-blue': selectedNavText === 'text-item-3' }">{{
+              i18n.t('repositories.highlight_component.assigned_label') }}
+
+          </div>
+          <div @click="handleSideNavClick" id="text-item-4" class="hover:cursor-pointer text-sn-grey"
+            :class="{ 'text-sn-science-blue': selectedNavText === 'text-item-4' }">{{
+              i18n.t('repositories.highlight_component.QR_label') }}</div>
+        </div>
+      </div>
+    </div>
+    <div id="highlight-container" class="w-[1px] h-[130px] flex flex-col justify-evenly bg-sn-light-grey">
+      <div v-for="(item, index) in itemsToCreate">
+        <div :key="index" :id="item" class="w-[5px] h-[28px] rounded-[11px]"
+          :class="{ 'bg-sn-science-blue relative left-[-2px]': item === selectedNavIndicator }"></div>
+      </div>
+    </div>
+  </div>
+</template>
+  
+<script>
+export default {
+  name: 'HighlightComponent',
+  props: {
+    itemsToCreate: Array,
+  },
+  data() {
+    return {
+      rootContainerEl: null,
+      selectedNavText: null,
+      selectedNavIndicator: null
+    }
+  },
+  created() {
+    this.rootContainerEl = this.$parent.$refs.wrapper
+  },
+  methods: {
+    handleSideNavClick(e) {
+      if (!this.rootContainerEl) {
+        return
+      }
+      let refToScrollTo
+      const targetId = e.target.id
+      switch (targetId) {
+        case 'text-item-1':
```

Comment:
This should be implemented using an array of objects that are passed to the componengt, not hardcoded. Also rename the component to Scrollspy, as this is the standard name for this thing.